pkg/pipelines/init.go

```diff
@@ -0,0 +1,253 @@
+	baseDir           = "base"
+	appsDir           = "apps"
+	kustomize         = "kustomization.yaml"
+	namespacesPath    = "01-namespaces/cicd-environment.yaml"
+	rolesPath         = "02-rolebindings/pipeline-service-role.yaml"
+	rolebindingsPath  = "02-rolebindings/pipeline-service-rolebinding.yaml"
+	secretsPath       = "03-secrets/github-webhook-secret.yaml"
+	tasksPath         = "04-tasks/deploy-from-source-task.yaml"
+	ciPipelinesPath   = "05-pipelines/ci-dryrun-from-pr-pipeline.yaml"
+	cdPipelinesPath   = "05-pipelines/cd-deploy-from-push-pipeline.yaml"
+	prBindingPath     = "06-bindings/github-pr-binding.yaml"
+	pushBindingPath   = "06-bindings/github-push-binding.yaml"
+	prTemplatePath    = "07-templates/ci-dryrun-from-pr-template.yaml"
+	pushTemplatePath  = "07-templates/cd-deploy-from-push-template.yaml"
+	eventListenerPath = "08-eventlisteners/cicd-event-listener.yaml"
+	routePath         = "09-routes/github-webhook-event-listener.yaml"
+)
+
+// Init function will initialise the gitops directory
+func Init(o *InitParameters) error {
+
+	if !o.SkipChecks {
+		installed, err := checkTektonInstall()
+		if err != nil {
+			return fmt.Errorf("failed to run Tekton Pipelines installation check: %w", err)
+		}
+		if !installed {
+			return errors.New("failed due to Tekton Pipelines or Triggers are not installed")
+		}
+	}
+
+	namespaces := namespaceNames(o.Prefix)
+
+	gitopsName := getGitopsRepoName(o.GitOpsRepo)
+	gitopsPath := filepath.Join(o.Output, gitopsName)
+
+	// check if the gitops dir already exists
+	exists, _ := isExisting(gitopsPath)
+	if exists {
+		return fmt.Errorf("%s already exists at %s", gitopsName, gitopsPath)
+	}
+
+	// key: path of the resource
+	// value: YAML content of the resource
+	outputs := map[string][]interface{}{}
+
+	if o.GitHubWebhookSecret != "" {
+		githubSecret, err := createOpaqueSecret(meta.NamespacedName(namespaces["cicd"], eventlisteners.GithubWebhookSecret), o.GitHubWebhookSecret, eventlisteners.WebhookSecretKey)
+		if err != nil {
+			return fmt.Errorf("failed to generate GitHub Webhook Secret: %w", err)
+		}
+
+		outputs[secretsPath] = append(outputs[secretsPath], githubSecret)
+	}
+
+	// create gitops pipeline
+	files := createPipelineResources(outputs, namespaces, o.GitOpsRepo, o.Prefix)
+
+	pipelinesPath := getPipelinesDir(gitopsPath, o.Prefix)
+
+	fileNames, err := writeResources(pipelinesPath, files)
+	if err != nil {
+		return err
+	}
+
+	sort.Strings(fileNames)
+	// kustomize file should refer all the pipeline resources
+	if err := addKustomize("resources", fileNames, filepath.Join(pipelinesPath, kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	if err := addKustomize("bases", []string{"./pipelines"}, filepath.Join(getCICDDir(gitopsPath, o.Prefix), kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	if err := addKustomize("bases", []string{}, filepath.Join(gitopsPath, envsDir, baseDir, kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	fmt.Println("\nGitops initialised")
```

Comment:
I think we can get rid of this to make it less verbose.

Reply:
Removed the print statement